pkg/comp-functions/functions/vshnkeycloak/ingress.go

```diff
@@ -0,0 +1,94 @@
+package vshnkeycloak
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	xfnproto "github.com/crossplane/function-sdk-go/proto/v1beta1"
+	xhelmv1 "github.com/vshn/appcat/v4/apis/helm/release/v1beta1"
+	vshnv1 "github.com/vshn/appcat/v4/apis/vshn/v1"
+	"github.com/vshn/appcat/v4/pkg/comp-functions/functions/common"
+	"github.com/vshn/appcat/v4/pkg/comp-functions/runtime"
+)
+
+// AddIngress adds an inrgess to the Keycloak instance.
+func AddIngress(_ context.Context, svc *runtime.ServiceRuntime) *xfnproto.Result {
+
+	comp := &vshnv1.VSHNKeycloak{}
+
+	err := svc.GetObservedComposite(comp)
+	if err != nil {
+		return runtime.NewFatalResult(fmt.Errorf("cannot get composite: %w", err))
+	}
+
+	if comp.Spec.Parameters.Service.FQDN == "" {
+		return nil
+	}
+
+	values, err := common.GetDesiredReleaseValues(svc, comp.GetName()+"-release")
+	if err != nil {
+		return runtime.NewWarningResult(fmt.Sprintf("cannot get desired release values: %s", err))
+	}
+
+	svc.Log.Info("Enable ingress for release")
+	enableIngresValues(comp, values)
+
+	release := &xhelmv1.Release{}
+	err = svc.GetDesiredComposedResourceByName(release, comp.GetName()+"-release")
+	if err != nil {
+		return runtime.NewWarningResult(fmt.Sprintf("cannot get desired release: %s", err))
+	}
+
+	vb, err := json.Marshal(values)
+	if err != nil {
+		return runtime.NewWarningResult(fmt.Sprintf("cannot marhal values: %s", err))
+	}
+
+	release.Spec.ForProvider.Values.Raw = vb
+
+	err = svc.SetDesiredComposedResourceWithName(release, comp.GetName()+"-release")
+	if err != nil {
+		return runtime.NewWarningResult(fmt.Sprintf("cannot set desired release: %s", err))
+	}
+
+	return nil
+}
+
+func enableIngresValues(comp *vshnv1.VSHNKeycloak, values map[string]any) {
+	fqdn := comp.Spec.Parameters.Service.FQDN
+
+	if fqdn != "" {
```

Comment:
Didn't you already tested for empty value in line 25 ?

Reply:
hmmm... yeah I guess it's redudant here. I'll remove it.